cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -581,10 +581,10 @@ public QueryHandle getTables(String catalog, String schemaPattern, String tableN
       SessionHandle sessionHandle = null;
       OperationHandle operationHandle = null;
       Map<String, String> sessionConf = startSession();
+      String database = getHiveDatabase(schemaPattern);
```

Comment:
It's still within an outer try-catch block, and the exception will get handled in the same way, except that no query handle will be closed (which is fine, since the query handle isn't created/opened yet).